lib/MusicBrainz/Server/Data/Tag.pm

```diff
@@ -120,6 +120,55 @@ sub get_cloud
     return $data;
 }
 
+sub rename_for_user {
+    my ($self, $tag_id, $editor_id, $new_tags) = @_;
+    my @new_tag_ids;
+
+    for my $new_tag (@$new_tags) {
+        my $tag_entity = $self->get_by_name($new_tag);
+        if ($tag_entity) {
+            push @new_tag_ids, $tag_entity->id;
+        } else {
+            my $new_tag_id = $self->sql->select_single_value(<<~'SQL', $new_tag);
+                INSERT INTO tag (name)
+                     VALUES (?)
+                  RETURNING id
+                SQL
+            push @new_tag_ids, $new_tag_id;
+        }
+    }
+
+    for my $entity_type (entities_with('tags')) {
+        my $entity_ids = $self->sql->select_single_column_array(<<~"SQL", $editor_id, $tag_id);
+            DELETE FROM ${entity_type}_tag_raw
+                  WHERE editor = ? AND tag = ? AND is_upvote = TRUE
+              RETURNING $entity_type
+            SQL
+
+        for my $new_tag_id (@new_tag_ids) {
```

Comment:
Paranoid question: are we worried that allowing to split 1 tag into N tags all linked to the same entities makes for a very efficient way to amplify data and/or spam the database? Is the example given ("australian hip-hop" -> "australian", "hip hop") more than just a very niche use case? Perhaps we should restrict the splitting to max N = 2, but even that seems iffy to me.

Reply:
There's a fair amount of tags around that could be split into genre tags (or even better, automatically "split" into "australian hip hop, hip hop" :) ). There's some risk of spam, but people can already submit tags using the API in large amounts, and we can already ban them if they seem to be abusing the system, so I'm not too worried? Any bad actors would still have to have added at least one tag to all the entities they want to spam anyway, and by that time, why wouldn't they just add all the tags there?

Reply:
IIUC this API endpoint doesn’t have any limit on the number of entities and tags that can be submitted at once. Or is there a size limit on POST requests received by MB Server?

Does it pose any question about service availability beyond the potential spam issue?

This patch would reproduce the current webservice issues into the website, making it even wider.